azure-pipelines.yaml

```diff
@@ -0,0 +1,121 @@
+trigger:
+  - master
+pr:
+  - none
+
+parameters:
+  - name: Run_Stage
+    displayName: Stage to Run
+    type: string
+    default: 'Deploy_Agent'
+    values:
+    - Deploy_Agent
+    - Destroy_Agent
+
+  - name: environment
+    displayName: Environment (sandbox default)
+    type: string
+    default: sandbox # USE upper and lower
+    values:
+      - sandbox
+ #     - upper
+ #     - lower
+
+variables:
+  - group: vh-azure-devops-agent
+  - name: TF_VAR_RG_NAME
+    value: "vh-devops-agent-rg-${{ parameters.environment }}"
+  - name: TF_VAR_KV_NAME
+    value: "vhdoagentkv${{ parameters.environment }}"
+  - name: TF_VAR_VM_NAME
+    value: "vh-devops-agent-vm-${{ parameters.environment }}"
+  - name: TF_VAR_NSG_NAME
+    value: "vh-devops-vm-nsg-${{ parameters.environment }}"
+  - name: TF_VAR_VNET_NAME
+    value: "vh-devops-vnet-${{ parameters.environment }}"
+
+stages:
```

Comment:
should there be a CI stage for validation

https://github.com/hmcts/azure-devops-templates/blob/master/templates/tests/terraform/allTests.yaml